### ISSUE TITLE
Certain viewing angle makes entities disappear

### ISSUE THREAD

**Issue by Zefferis** [sp614x/optifine#2088]
When the upward viewing degree is between ``~42.6 to ~50.6`` (depending on distance) mob entities disappear.

This is an odd experience when you're looking around and very close to entities like passive animals, or facing upwards when fighting mobs.

This may be related to https://github.com/sp614x/optifine/issues/333 #333

**Comment by sp614x** [comment]
Can you post:
- F3 screenshot
- screenshot with visible entities (viewing angle below 43)
- screenshot with invisible entities (same position/direction, angle above 42)

**Comment by Zefferis** [comment]
![2018-12-07_17 02 00](https://user-images.githubusercontent.com/15270437/49679669-ecf38280-fa41-11e8-9156-e27e482540f5.png)
![2018-12-07_17 02 03](https://user-images.githubusercontent.com/15270437/49679670-ecf38280-fa41-11e8-9b5c-2a3fba757f31.png)
![2018-12-07_17 02 06](https://user-images.githubusercontent.com/15270437/49679671-ecf38280-fa41-11e8-922f-29bb2ce21951.png)


![2018-12-07_17 02 25](https://user-images.githubusercontent.com/15270437/49679674-faa90800-fa41-11e8-9c51-a03ca79a2dc4.png)
![2018-12-07_17 02 26](https://user-images.githubusercontent.com/15270437/49679675-faa90800-fa41-11e8-8400-de9f2f8074ea.png)
![2018-12-07_17 02 28](https://user-images.githubusercontent.com/15270437/49679676-faa90800-fa41-11e8-8160-cfd75e7879a9.png)

**Comment by Zefferis** [comment]
Yep, it's not a major issue; it's just that when your character is very close by, they pop back into view when you look a little higher. 

I was thinking this is just an effect of not rendering something that's off-screen, but it popping back into view doesn't make that seem to be the case since it should stay not rendered (or so I think 🤷‍♂️ )

**Comment by sp614x** [comment]
Can't reproduce it currently.

**Comment by Zefferis** [comment]
Uhhh, So I loaded up those entities without using Optifine, and it seems that mobs that are on y level 63 have this kind of effect; is that the same behavior for you? (My coords have me at y=63)

Doesn't seem to happen above 63 or below 63 (as far as I can)

Because it happened outside of Optifine it isn't an Optifine issue, but it's a strange rendering issue in MC

**Comment by TheSnowyChickens** [comment]
This issue occures with 1.8+, in 1.7.10 all was fine. So, thats just a "bug" from MC.

**Comment by Zefferis** [comment]
Interesting @TheSnowyChickens ; is it only effecting mobs at y=63 ?

**Comment by sp614x** [comment]
Reproduced on y=63 and it also happens in vanilla:

![image](https://user-images.githubusercontent.com/16635441/49749155-dbabaf80-fca7-11e8-818d-b127c33a55a4.png)

![image](https://user-images.githubusercontent.com/16635441/49749187-eb2af880-fca7-11e8-92c1-5db95a671734.png)

**Comment by sp614x** [comment]
The reason is that y=63 is the last Y level of the cubic chunk (F3+G to see chunk borders). 
The entity belongs to the lower cubic chunk (feet pos), but it's head is in the upper cubic chunk.
The entities are rendered together with the chunks that they belong to. When the lower shunk is not renderer because it is outside of the view frustum the entity is also not rendered, even though its upper part is still visible.

**Comment by TheSnowyChickens** [comment]
Immersive addon Steam Engine. (yes, Opti is installed, but it also happens without)
![2018-12-10_18 28 11](https://user-images.githubusercontent.com/29357962/49749774-67720b80-fca9-11e8-9f2f-5e6487fbd581.png)

**Comment by sp614x** [comment]
The bug is probably existing since 1.8 when the optimized view culling was added.
Definitely a bug, but not easy to fix without hurting performance.

**Comment by TheSnowyChickens** [comment]
Oh.

**Comment by sp614x** [comment]
Probably related: https://bugs.mojang.com/browse/MC-65040
